src/loaders/configure/nonce.js

```diff
@@ -2,7 +2,7 @@
 
 __webpack_require__.nc = (() => {
   try {
-    if (document.currentScript && document.currentScript.nonce) {
+    if (document && document.currentScript && document.currentScript.nonce) {
       return document.currentScript.nonce
     }
   } catch (ex) {
```

Comment:
Can't this whole try-catch just be the undefined-err proof: `return typeof document !== 'undefined' && document.currentScript?.nonce` ?

Reply:
Wasnt sure if Webpack transpiles these native overrides, since this was the way Patrick wrote this method.  The optional chaining would break IE if not.

plus your example would return `false` instead of `undefined` for cases where document is not defined which i'm also not sure what webpack would do with when adding the nonce to scripts

Reply:
shared in slack